src/ui/win32/bindings/MemoryViewerControlBinding.cpp

```diff
@@ -231,6 +231,42 @@ bool MemoryViewerControlBinding::HandleNavigation(UINT nChar)
             }
             return true;
 
+        case VK_ADD:
+        case VK_SUBTRACT:
+        {
+            auto nAddress = m_pViewModel.GetAddress();
+            const auto& pEmulatorContext = ra::services::ServiceLocator::Get<ra::data::context::EmulatorContext>();
+            auto iMem = pEmulatorContext.ReadMemory(m_pViewModel.GetAddress(), m_pViewModel.GetSize());
+            auto iMaxValue = (1ULL << (ra::data::MemSizeBytes(m_pViewModel.GetSize()) * 8)) - 1;
+            auto iModifier = 0x1;
+
+            if (bControlHeld)
+                iModifier *= 4;
+            if (bShiftHeld)
+                iModifier *= 8;
+
+            // Min and Max tresholds
+            if ((iMem >= iMaxValue and nChar == VK_ADD) or (iMem == 0x0 and nChar == VK_SUBTRACT))
+                return false;
+
+            // Set value to min or max when tresholds are exceeded
+            if ((iMaxValue - iMem < iModifier) and nChar == VK_ADD)
+                iModifier = (iMaxValue - iMem);
+            else if ((iMem < iModifier) and nChar == VK_SUBTRACT)
+                iModifier = iMem;
+
+            nChar == VK_SUBTRACT ? iMem -= iModifier : iMem += iModifier;
```

Comment:
instead of checking nChar multiple times, suggest regrouping into a logical if (or even splitting the cases):
```
if (nChar == VK_SUBTRACT)
{
    if (iMem == 0)
        return false;
    if (iMem < iModifier)
        iMem = 0;
    else
        iMem -= iModifier;
}
else
{
   ...
}